new/language/implementation/javascript/javascript.go

```diff
@@ -0,0 +1,240 @@
+package javascript
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/smacker/go-tree-sitter/javascript"
+	"github.com/ssoroka/slice"
+	"golang.org/x/exp/slices"
+
+	"github.com/bearer/curio/new/language/implementation"
+	"github.com/bearer/curio/new/language/tree"
+	"github.com/bearer/curio/pkg/util/regex"
+
+	patternquerytypes "github.com/bearer/curio/new/language/patternquery/types"
+	sitter "github.com/smacker/go-tree-sitter"
+)
+
+var (
+	variableLookupParents = []string{"pair", "arguments", "binary_expression"}
+
+	anonymousPatternNodeParentTypes = []string{}
+	patternMatchNodeContainerTypes  = []string{}
+
+	// $<name:type> or $<name:type1|type2> or $<name>
+	patternQueryVariableRegex = regexp.MustCompile(`\$<(?P<name>[^>:!]+)(?::(?P<types>[^>]+))?>`)
+	allowedPatternQueryTypes  = []string{"identifier", "property_identifier", "_", "member_expression"}
```

Comment:
Do we need `member_expression` here? I think it's better not to, because of the way things nest. eg. a user might like to do `x.$<VAR>` where `VAR` is filtered on `a.b` but it won't actually match `x.a.b` as it's nested like `(x.a).b` in the AST.